play-services-location/core/src/main/kotlin/org/microg/gms/location/manager/LocationManager.kt

```diff
@@ -277,34 +282,24 @@ class LocationManager(private val context: Context, override val lifecycle: Life
     }
 
     private suspend fun requestPermission(permissions: List<String>): Boolean {
-        val (completable, deferred) = activePermissionRequestLock.withLock {
-            if (activePermissionRequest == null) {
-                val completable = CompletableDeferred<Boolean>()
-                activePermissionRequest = completable
-                completable to activePermissionRequest!!
-            } else {
-                null to activePermissionRequest!!
-            }
-        }
-        if (completable != null) {
-            val intent = Intent(context, AskPermissionActivity::class.java)
-            intent.putExtra(EXTRA_MESSENGER, Messenger(object : Handler(Looper.getMainLooper()) {
-                override fun handleMessage(msg: Message) {
-                    if (msg.what == Activity.RESULT_OK) {
-                        lastLocationCapsule.fetchFromSystem()
-                        onRequestManagerUpdated()
-                        val grantResults = msg.data?.getIntArray(EXTRA_GRANT_RESULTS) ?: IntArray(0)
-                        completable.complete(grantResults.size == permissions.size && grantResults.all { it == PackageManager.PERMISSION_GRANTED })
-                    } else {
-                        completable.complete(false)
-                    }
+        val permissionRequest = activePermissionRequestLock.withLock {CompletableDeferred<Boolean>()}
```

Comment:
This can cause the activity to be invoked multiple times, we only want to start it once